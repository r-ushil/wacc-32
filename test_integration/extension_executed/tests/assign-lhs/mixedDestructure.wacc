
begin
  struct IntBox {
    int i
  }
  struct IntsBox {
    int[] ns
  }

  # Array to modify.
  int[] x = [1,2,3,4];

  # Struct to destructure.
  IntBox ib = IntBox { i: 42 };

  # Destructure into array.
  IntBox { i: x[2] } = ib;

  println "x:";
  println x[0];
  println x[1];
  println x[2];
  println x[3];

  IntsBox IntsBox { ns: [a, b, c, d] } = IntsBox { ns: x };

  println "x destructured via ints box:";
  println a;
  println b;
  println c;
  println d; 

  pair(IntBox, pair(int, char[])) newpair(
    IntBox { i: a1 },
    newpair(
      b1,
      [l1, l2, l3, l4]
    )
  ) = newpair(
    IntBox { i: 1 },
    newpair(
      5,
      ['a', 'b', 'c', 'd']
    )
  );

  println "complex destructure declare:";
  println a1; #Â 1
  println b1; # 5
  println l1; # a
  println l2; # b
  println l3; # c
  println l4; # d

  skip
end