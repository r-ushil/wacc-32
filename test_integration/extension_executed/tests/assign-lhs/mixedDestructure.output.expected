Compiler exit code: 0
--------- stdout ---------
x:
1
2
42
4
x destructured via ints box:
1
2
42
4
complex destructure declare:
1
5
a
b
c
d
--------------------------
Exit code: 0
